drivers/dma/Kconfig.stm32

```diff
@@ -0,0 +1,46 @@
+# Kconfig - DMA configuration options
+#
+#
+# Copyright (c) 2016 Intel Corporation
+# Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig DMA_STM32
+	bool "Enable STM32 DMA driver"
+	select DYNAMIC_INTERRUPTS
+	depends on SOC_FAMILY_STM32
+	depends on HEAP_MEM_POOL_SIZE != 0
+	help
+	  DMA driver for STM32 series SoCs.
+
+if DMA_STM32
+
+config DMA_1
+	bool "Enable STM32 DMA Controller 1"
+	help
+	  Enable support for DMA1 in the driver.
+	  Say y here if you want to use DMA1 device.
+
+config DMA_2
+	bool "Enable STM32 DMA Controller 2"
+	help
+	  Enable support for DMA2 in the driver.
+	  Say y here if you want to use DMA2 device.
+
+config DMA_STM32_V1
+	bool
+	depends on SOC_SERIES_STM32F2X || SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
```

Comment:
Do we keep F7 for now?

Reply:
I would like to keep it, I was thinking there maybe other people would like to help solving the buffer problem and even if the code does not work on f7 it does not breaks anything since there wasn't any support for the f7. But pushing code cannot pass the test is not ok, right? If it is not ok I'll remove this.